src/prefect/utilities/executors.py

```diff
@@ -108,20 +110,61 @@ def error_handler(signum, frame):  # type: ignore
         signal.alarm(0)
 
 
+def multiprocessing_safe_retrieve_value(
+    *args: Any,
+    _container: multiprocessing.Queue,
+    _ctx_dict: dict,
+    _fn_pkl: bytes,
+    **kwargs: Any
+) -> None:
+    """
+    Gets the return value from a function and puts it in a multiprocessing-safe container
+    Helper function for `multiprocessing_timeout`
+
+    Passing the function pre-serialized allows us to escape the limitations of the
+    python native pickler which will fail on tasks defined in scripts because of
+    a name mismatch.
+
+    Args:
+        - *args: Arguments to pass to the function
+        - **kwargs: Keyword arguments to pass to the function
+        - _container (multiprocessing.Queue): A multiprocessing queue for the result
+        - _ctx_dict (dict): The prefect.context dictionary
+        - _fn_pkl (bytes): A `cloudpickle` serialized copy of the function to call
+
+    Returns:
+        - None -- passes the return value or exception into the queue.
+        Callers are expected to re-raise any exceptions.
+    """
+    try:
+        with prefect.context(_ctx_dict):
+            fn = cloudpickle.loads(_fn_pkl)
+            # We could check if the `fn` is callable here but it will get caught by
+            # the try-except anyway. If the given error message is confusing, we
+            # can do an explicit check and provide more details
+            val = fn(*args, **kwargs)
+        _container.put(val)
```

Comment:
We should also use `cloudpickle.dumps` on the result (or the exception), as not all results may be pickleable either.

Reply:
Good idea! Done.